packages/engine/lib/error/src/macros.rs

```diff
@@ -5,20 +5,18 @@ pub mod __private {
     use crate::Report;
 
     pub mod kinds {
-        use core::marker::PhantomData;
+        #![allow(clippy::unused_self)]
 
-        use crate::{Message, Report};
+        use crate::{Context, Message, Report};
 
-        pub trait AdhocKind: Sized {
-            fn __kind(&self) -> Adhoc {
-                Adhoc
+        pub trait MessageTag {
```

Comment:
I will add documentation on those "black magic" when this got cleaned up. I expect changes here in a follow-up PR.